tests/testlib/s2n_testlib.h

```diff
@@ -244,3 +244,10 @@ extern const s2n_parsed_extension EMPTY_PARSED_EXTENSIONS[S2N_PARSED_EXTENSIONS_
 int s2n_kem_recv_public_key_fuzz_test(const uint8_t *buf, size_t len, struct s2n_kem_params *kem_params);
 int s2n_kem_recv_ciphertext_fuzz_test(const uint8_t *buf, size_t len, struct s2n_kem_params *kem_params);
 int s2n_kem_recv_ciphertext_fuzz_test_init(const char *kat_file_path, struct s2n_kem_params *kem_params);
+
+/* A collection of callbacks run during inet socket self talk tests */
+struct self_talk_inet_socket_callbacks {
+    S2N_RESULT (*server_post_handshake_cb)(struct s2n_connection *conn);
+    S2N_RESULT (*client_post_handshake_cb)(struct s2n_connection *conn);
+};
+S2N_RESULT s2n_noop_inet_socket_cb(struct s2n_connection *conn);
```

Comment:
they are building blocks for later [PRs](https://github.com/toidiu/s2n-tls/blob/b7aa689c11898df970f670c851e4b8e0a262991e/tests/unit/s2n_self_talk_inet_socket_test.c#L233-L289) where we enable ktls selectively on send/recv

Reply:
Right ok - the callback struct itself makes sense. But even in that example, it seems to only be used in `s2n_self_talk_inet_socket_test`. If the callback struct isn't needed outside of this test, couldn't it just be defined in the test itself rather than testlib?

Reply:
Ahh, yea the missing piece is that there might be more than 1 inet_test file.

This one tests happy path and there are other that test alerts, or not happy path. But I guess for the purposes of this PR we can move it to the same test file. Also putting impls of the callback `s2n_noop_inet_socket_cb` in a common place makes the tests less cluttered.